tensorflow_datasets/core/example_serializer_test.py

```diff
@@ -218,6 +218,36 @@ def test_item_to_tf_feature_string_check(self):
     ):
       example_serializer._item_to_tf_feature(example_item, tensor_info)
 
+  def test_dict_to_tf_example_error_reraise(self):
+    # Test error reraise in _dict_to_tf_example.
+    example_data = {'input': [1, 2, 3]}
+    tensor_info = {
+        'input': feature_lib.TensorInfo(
+            shape=(2,),
+            dtype=tf.int64,
+        ),
+    }
+    with self.assertRaisesRegex(
+        ValueError,
+        '^Error while serializing feature `input`:(.*)',
+    ):
+      example_serializer._dict_to_tf_example(example_data, tensor_info)
+
+  def test_dict_to_tf_example_flatten_nest_dict(self):
+    # Test that a KeyError from zip_dict in _dict_to_tf_example is not reraised.
+    example_data = {'x': [1, 2, 3]}
+    tensor_info = {
+        'y': feature_lib.TensorInfo(
+            shape=(2,),
+            dtype=tf.int64,
+        ),
+    }
+    with self.assertRaisesRegex(
+        KeyError,
+        '^(?!Error while serializing feature)(.*)',
```

Comment:
Note: No need to have a full regex which are difficult to parse.

`self.assertRaisesRegex(KeyError, 'Error while serializing feature')` works well enough and the likelihood that another exception match is very low.

Reply:
Got it. Fixed in 72c8d5adffe5b58d291da3bca50c4251e1b8f2bf.